modules/graph-data-science/pages/node-classification.adoc

```diff
@@ -0,0 +1,481 @@
+MERGE (c)-[:APPEARED_IN]->(co);
+
+LOAD CSV WITH HEADERS FROM "https://raw.githubusercontent.com/tomasonjo/blog-datasets/main/Marvel/heroToEvent.csv" as row
+MATCH (c:Character{id:row.hero})
+MATCH (e:Event{id:row.event})
+MERGE (c)-[:PART_OF_EVENT]->(e);
+
+LOAD CSV WITH HEADERS FROM "https://raw.githubusercontent.com/tomasonjo/blog-datasets/main/Marvel/heroToGroup.csv" as row
+MATCH (c:Character{id:row.hero})
+MATCH (g:Group{id:row.group})
+MERGE (c)-[:PART_OF_GROUP]->(g);
+
+LOAD CSV WITH HEADERS FROM "https://raw.githubusercontent.com/tomasonjo/blog-datasets/main/Marvel/heroToHero.csv" as row
+MATCH (s:Character{id:row.source})
+MATCH (t:Character{id:row.target})
+CALL apoc.create.relationship(s,row.type, {}, t) YIELD rel
+RETURN distinct 'done';
+
+LOAD CSV WITH HEADERS FROM "https://raw.githubusercontent.com/tomasonjo/blog-datasets/main/Marvel/heroStats.csv" as row
+MATCH (s:Character{id:row.hero})
+CREATE (s)-[:HAS_STATS]->(stats:Stats)
+SET stats += apoc.map.clean(row,['hero'],[]);
+
+LOAD CSV WITH HEADERS FROM "https://raw.githubusercontent.com/tomasonjo/blog-datasets/main/Marvel/heroFlight.csv" as row
+MATCH (s:Character{id:row.hero})
+SET s.flight = row.flight;
+
+MATCH (s:Stats)
+WITH keys(s) as keys LIMIT 1
+MATCH (s:Stats)
+UNWIND keys as key
+CALL apoc.create.setProperty(s, key, toInteger(s[key]))
+YIELD node
+RETURN distinct 'done';
+----
+
+This query creates a series of nodes and their labels and properties: `Comic`, `Character`, `Stats` along with a variety of edges such as which comics the characters appeared in which comics, who is an enemby of whom, etc.  You can see a schema of this graph using `CALL db.schema.visualization()` here:
+
+image::marvel-schema.png[Marvel Universe schema, link="{imagesdir}/marvel-schema.png",role="popup-link"]
+
+(Note that for the simplicity of the image we are not showing the node properties since there are many.)
+
+We next bring in the character traits from the stats to be node properties:
+
+[source, cypher]
+----
+MATCH (c:Character)-[:HAS_STATS]->(s)
+WITH c, s.strength as strength, s.fighting_skills as fighting_skills, s.durability as durability, s.speed as speed, s.intelligence as intelligence, s.energy as energy
+SET c.strength=strength,
+    c.fighting_skills=fighting_skills,
+    c.durability=durability,
+    c.speed=speed,
+    c.intelligence=intelligence,
+    c.energy=energy
+RETURN count(c)
+----
+
+These node properties will eventually be used to build the node classification model via both embeddings as well as the tabular approach.
+
+Next, we set up the co-occurance of characters such that we can identify which characters appear with which other characters and how often (which will be used to identify the edge weighting).  This is done via:
+
+[source, cypher]
+----
+MATCH (c1:Character)-[:APPEARED_IN]->(c:Comic)<-[:APPEARED_IN]-(c2:Character) 
+WITH c1, c2, count(c) as weight
+MERGE (c1)-[:APPEARED_WITH{times:weight}]->(c2)
+MERGE (c2)-[:APPEARED_WITH{times:weight}]->(c1)
```

Comment:
I don't think you need to have this relationship twice. You can rather treat it as UNDIRECTED when loading into GDS

Reply:
This is what is in Alicia's demo.  So I am not planning on changing any of those queries (with the exception of the one hot encoding not being used in anything downstream of it) since they are straight out of her repo.